fastestimator/schedule/lr_shedule.py

```diff
@@ -59,3 +71,125 @@ def cosine_decay(time: int,
+        model.add(tf.keras.layers.MaxPooling1D(pool_size=2))
+        model.add(tf.keras.layers.LSTM(64, return_sequences=True))
+        model.add(tf.keras.layers.LSTM(64))
+        model.add(tf.keras.layers.Dense(32, activation='relu'))
+        model.add(tf.keras.layers.Dense(3, activation='softmax'))
+        return model
+
+    def accumulate_single_train_loss(self, train_loss: float) -> None:
+        self.train_loss_one_cycle.append(train_loss)
+
+    def accumulate_single_eval_loss(self, eval_loss: float) -> None:
+        self.eval_loss_one_cycle.append(eval_loss)
+
+    def accumulate_all_lrs(self, lr: float) -> None:
+        self.all_train_lr.append(lr)
+
+    def gather_multiple_eval_losses(self) -> None:
+        self.all_eval_loss.append(self.eval_loss_one_cycle)
+        self.eval_loss_one_cycle = []
+
+    def gather_multiple_train_losses(self) -> None:
+        self.all_train_loss.append(self.train_loss_one_cycle)
+        self.train_loss_one_cycle = []
+
+    def predict_next_multiplier(self) -> float:
+        train_loss, missing = self._merge_list(self.all_train_loss)
+        train_loss = self._preprocess_train_loss(train_loss, missing)
+        val_loss, missing = self._merge_list(self.all_eval_loss)
+        val_loss = self._preprocess_val_loss(val_loss, missing)
+        train_lr, missing = self._merge_list(self.all_train_lr)
+        train_lr = self._preprocess_train_lr(train_lr, missing)
+        model_inputs = np.concatenate((train_loss, val_loss, train_lr), axis=1)
+        model_inputs = np.expand_dims(model_inputs, axis=0)
+        with tf.device("cpu:0"):
+            model_pred = self.model(model_inputs, training=False)
+        action = np.argmax(model_pred)
+        return self.lr_multiplier[action]
+
+    def _preprocess_val_loss(self, val_loss: List[float], missing: int) -> np.ndarray:
+        if val_loss:
+            target_size = (3 - missing) * 100
+            val_loss = np.array(val_loss, dtype="float32")
+            val_loss = zscore(val_loss)
+            val_loss = cv2.resize(val_loss, (1, target_size), interpolation=cv2.INTER_NEAREST)
+            if val_loss.size < 300:
+                val_loss = np.pad(val_loss, ((300 - val_loss.size, 0), (0, 0)), mode='constant', constant_values=0.0)
+        else:
+            val_loss = np.zeros([300, 1], dtype="float32")
+        return val_loss
+
+    def _preprocess_train_lr(self, train_lr: List[float], missing: int) -> np.ndarray:
+        target_size = (3 - missing) * 100
+        train_lr = np.array(train_lr) / train_lr[0]
+        train_lr = cv2.resize(train_lr, (1, target_size), interpolation=cv2.INTER_NEAREST)
+        if train_lr.size < 300:
+            train_lr = np.pad(train_lr, ((300 - train_lr.size, 0), (0, 0)), mode='constant', constant_values=1.0)
+        return train_lr
+
+    def _preprocess_train_loss(self, train_loss: List[float], missing: int) -> np.ndarray:
```

Comment:
can't we combine preprocess_train_loss and preprocess_val_loss?

Reply:
preprocessing val loss is still different from train. for example, prepcoessing val loss has an extra step of filling in zero data. 

Also, keeping them separate helps with experimenting different train/val loss processing in the future.